app/processing/apply-dual-accounting.js

```diff
@@ -0,0 +1,28 @@
+const { FDMR, BPS } = require('../constants/schemes')
+const { DOM00, DOM01, DOM10 } = require('../constants/domestic-fund-codes')
+
+const addDualAccounting = (paymentRequests, previousPaymentRequests) => {
+  for (const paymentRequest of paymentRequests) {
+    for (const invoiceLine of paymentRequest.invoiceLines) {
+      if (paymentRequest.schemeId === FDMR || paymentRequest.schemeId === BPS) {
+        if (paymentRequest.marketingYear >= 2020) {
+          invoiceLine.fundCode = DOM10
+        } else {
+          if (previousPaymentRequests.length) {
+            const previousFundCode = previousPaymentRequests[previousPaymentRequests.length - 1].invoiceLines[0].fundCode
```

Comment:
Thinking about this, there are scenarios where the previous payment request might not have any invoices lines as no further action was needed but the one before might.  

Suggest changing this to filter to the last payment request with invoice lines, then getting the fund code.

Reply:
I'm not too sure how I would go about filtering for that - are there any examples I could reference?

Reply:
You could do something like this:

```
const previousPaymentRequestsWithInvoiceLines = previousPaymentRequests.filter(x => x.invoiceLines.length)
const previousFundCode = previousPaymentRequestsWithInvoiceLines[previousPaymentRequestsWithInvoiceLines.length - 1]?invoiceLines[0].fundCode
```

Adding the `?` so it doesn't error if there aren't any.

Probably wise to add a test for this scenario to confirm it all works